### PR TITLE
Boost search results when the keyword matches certain fileds

### PR DESCRIPTION
**Pull request by dozhang**

### CODE REVIEW

File: zubbi/models.py [hunk 1]
```diff
@@ -39,6 +39,13 @@
     "verify_mode": "CERT_REQUIRED",
 }
 
+# We want to boost following fields
+SEARCH_BOOST_FIELDS = {
+    "role_name": "role_name^1.5",
+    "job_name": "job_name^1.5",
+    "description": "description^1.2",
```
Comment (felixedel) [inline zubbi/models.py:46]: Did you test if this works as expected or could you point me to some documentation? I just couldn't find anything about this syntax in regard to boosting, apart from an old Stackoverflow answer which points to an outdated documentation: https://stackoverflow.com/a/24524040

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Unreleased
+
+### Improvements
+- **UI:** Improve search result ordering. The search result will be boosted to
```
Comment (felixedel) [inline CHANGELOG.md:6]: As we just discussed, we should write something about priority here. Something like (maybe you find a better wording):
... The search results will be ordered by the priority of the fields matching the query. The priority is like the following name -> description -> tenant/repo.

File: zubbi/models.py [hunk 3]
```diff
@@ -193,9 +200,9 @@ def _select_doc_type(doc_type, block_class):
             return [block_class]
         return [AnsibleRole, ZuulJob]
 
-    def search_query(self, query, fields, exact=False, extra_filter=None):
-        # Elasticsearch cannot work with a set, only with list
```
Comment (felixedel) [inline zubbi/models.py:197]: Maybe it's worth to keep the comment as we still change the value to a list ;-)

**Review by felixedel — CHANGES_REQUESTED** [review]

File: zubbi/models.py (hunk 1 above)
Comment (dozhang) [reply zubbi/models.py:46]: I have tested it works as expected, the syntax is mentioned also here: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-multi-match-query.html

File: zubbi/models.py (hunk 3 above)
Comment (dozhang) [reply zubbi/models.py:197]: will fix.

File: CHANGELOG.md (hunk 2 above)
Comment (dozhang) [reply CHANGELOG.md:6]: will fix.

**Review by felixedel — APPROVED** [review]

**Pull request merged**